lib/apiservers/engine/backends/vicbackends.go

```diff
@@ -71,8 +76,13 @@ func Init(portLayerAddr, product string, config *config.VirtualContainerHostConf
 	portLayerClient = client.New(t, nil)
 	portLayerServerAddr = portLayerAddr
 
+	log.Info("Creating image store")
+	if err := createImageStore(); err == nil {
```

Comment:
That loop is running inside a goroutine and we don't want to run in a goroutine because the race condition will move (assuming we don't ditch the goroutine). Also, it seems like it would be kinda clunky to have to account for retrying image store creation or cache updating or both within the same loop, given that one could succeed while the other could fail. Your thoughts?